crates/router/src/core/errors/api_error_response.rs

```diff
@@ -143,6 +143,24 @@ pub enum ApiErrorResponse {
     },
 }
 
+#[derive(Clone)]
+pub enum NotImplementedMessage {
+    Reason(String),
+    Default,
+}
+
+impl std::fmt::Debug for NotImplementedMessage {
+    fn fmt(&self, fmt: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let message = match self {
+            Self::Reason(m) => format!("{} is not implemented", m),
+            Self::Default => {
+                "This API is under development and will be made available soon.".to_string()
+            }
+        };
+        write!(fmt, "{message}")
```

Comment:
This would create one more allocation than necessary. 

```suggestion
        match self {
            Self::Reason(message) => write!(fmt, "{message} is not implemented"),
            Self::Default => {
                write!(fmt, "This API is under development and will be made available soon.")
            }
        }
```